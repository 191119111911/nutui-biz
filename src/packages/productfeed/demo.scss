#app .demo {
  padding: 57px 0 0;
}
.nut-tabs__titles {
  background: #eee;
}
.nut-tabpane {
  padding: 0 20px 20px;
  background: #eee;
}
.nut-biz-productfeed {
  height: calc(100vh - 145px);
  width: 100%;
  padding: 0;
  overflow-y: auto;
  overflow-x: hidden;
  &::-webkit-scrollbar { width: 0 !important }
}
// 商品样式
.nut-biz-productfeedItem {
  margin-bottom: 6px;
  .img-label {
    width: 63px;
    height: 16px;
    img {
      display: block;
      width: 100%;
    }
  }
  .nut-image {
    img {
      width: 100%;
      border-radius: 8px;
    }
  }
  .name-box {
    margin-bottom: 7px;
    font-size: 14px;
    font-family: PingFang SC;
    font-weight: normal;
    color: rgba(0,0,0,0.85);
    display: -webkit-box;
    -webkit-box-orient: vertical;
    -webkit-line-clamp: 2;
    overflow: hidden;
    word-break: break-all;
    .label {
      display: inline-flex;
      align-items: center;
      height: 12px;
      margin-right: 5px;
      padding: 0 2px;
      font-size: 10px;
      font-family: PingFang SC;
      vertical-align: 2px;
      color: rgba(255,255,255,1);
      background: linear-gradient(135deg, rgba(255,64,79,1) 0%,rgba(250,44,25,1) 100%);
      border-radius: 2px;
    }
  }
  .price {
    font-size: 0;
    .nut-price {
      font-family: JDZH-Regular;
      color: rgba(250,44,25,1);
      .nut-price__symbol-large, .nut-price__decimal-large {
        font-size: 12px;
      }
      .nut-price__integer-large {
        font-size: 18px;
      }
    }
  }
}

.demo2 {
  .nut-biz-productfeedItem__content {
    // margin-left: 6px;
  }
  .name-box {
    height: 42px;
  }
  .desc-box {
    height: auto;
    margin-bottom: 10px;
    font-size: 12px;
    color: #8c8c8c;
  }
}