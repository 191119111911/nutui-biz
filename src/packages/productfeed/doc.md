#  ProductFeed 商品 Feed 流

### 介绍

商品 Feed 流组件可配置下拉刷新、列表加载、上滑加载功能，适用于商品信息展示，常见于页面的底部。

<!-- 应用场景包括图片分享、新闻推荐类、商品推荐等。 -->

### 安装
``` javascript
import { ProductFeed, ProductFeedItem } from '@nutui/nutui-biz';
```

## 代码演示

### 基本用法

:::demo

```ts
import  React from 'react';
import { Price } from '@nutui/nutui-react';
import { Card } from '@nutui/nutui-biz';

const App = () => {

  const [list1, setList1] = useState([] as any)

  const [hasMore1, setHasMore1] = useState(true)

  const data = [
    {
      imgUrl: '//img13.360buyimg.com/imagetools/jfs/t1/190855/7/12881/42147/60eb0cabE0c3b7234/d523d551413dc853.png',
      name: translated.name,
      desc: translated.desc,
      price: '388',
      vipPrice: '378',
    }, {
      imgUrl: '//img13.360buyimg.com/imagetools/jfs/t1/190855/7/12881/42147/60eb0cabE0c3b7234/d523d551413dc853.png',
      name: translated.name,
      desc: translated.desc,
      price: '388',
      vipPrice: '378',
    },
    ...
  ]

  const loadMore1 = (done: () => void) => {
    setTimeout(() => {
      const curLen = list1.length
      if (list1.length >= data.length) {
        setHasMore1(false)
      } else {
        for (let i = curLen; i < (curLen + 6 > data.length ? data.length : curLen + 6) ; i++) {
          list1.push(data[i])
        }
        setList1([...list1]) 
      }
      done()
    }, 500)
  }

  const refresh = (done: () => void) => {
    setTimeout(() => {
      console.log('刷新成功')
      done()
    }, 1000)
  }

  const init1 = () => {
    for (let i = 0; i < 6; i++) {
      list1.push(data[i])
    }
    setList1([...list1])
  }

  useEffect(() => {
    init1()
  }, [])

  return (
    <ProductFeed
      className="demo1"
      id="refreshScroll1"
      hasMore={hasMore1}
      containerId="refreshScroll1"
      useWindow={false}
      onLoadMore={loadMore1}
    >
      {list1.map((item: any, index: number)=> {
        return (
          <ProductFeedItem
            key={index}
            gutter={6}
            col={2}
            imgUrl={item.imgUrl}
            imgHeight="164"
            imgTag={<div className='img-label'><img src="https://img12.360buyimg.com/imagetools/jfs/t1/186347/7/7338/1009/60c0806bE0b6c7207/97fd04b48d689ffe.png" /></div>}
          >
            <>
              <div className="name-box">
                <div className="label">自营</div>
                {item.name}
              </div>
              <div className="bottom">
                <div className="price-box">
                  <div className="price">
                    <Price price={item.price} />
                  </div>
                </div>
              </div>
            </>
          </ProductFeedItem>
        )
      })}
    </ProductFeed>
  );
};
export default App;
```

:::


## API

### Props


| 字段    | 说明                                       | 类型    | 默认值    |
|---------|--------------------------------------------|---------|-----------|
| imgUrl   | 左侧图片Url                                 | String  | -         |
| title     | 标题                   | String  | -    |
| price | 商品价格                         | String  | -      |
| vipPrice     | 会员价格                               | String | -    |
| shopDesc  | 店铺介绍                                  | String | -    |
| delivery     | 配送方式 | String  | -      |
| shopName   | 店铺名称| String  | -      |
| prolistTpl   | 自定义商品介绍| React.ReactNode  | -      |
| priceTpl   | 价格自定义内容 | React.ReactNode  | -      |
| originTpl   | 价格后方自定义内容| React.ReactNode  | -      |
| shopTagTpl   | 店铺介绍自定义| React.ReactNode  | -      |
| footerTpl   | 右下角内容自定义| React.ReactNode  | -      |
| showType   | 展示形式，可选：`full-line`、`half-line`| String  | `full-line`      |
| bottomTpl   | 底部内容自定义| React.ReactNode  | -      |
| infotpl   | 信息内容自定义| React.ReactNode  | -      |
| isNeedPrice   | 是否需要价格展示| Boolean  | `true`      |
| imgTag   | 商品图片标签，常用于标志双 11、直播等| String  | -      |
| imgTagDirection   | 商品图片标签呈现位置，可选：`top-left`、`top-right`| String  | `top-left`      |
| isLazy   | 是否为懒加载图片| Boolean  | `false`      |
| titleTag   | 标题左侧标签，常用于活动标记等，不设置此参数或此参数为空，不展示| React.ReactNode  | -      |
| titleLine   | 标题行数，默认两行，想展示更多下面内容可设置为 1| String \| Number  | `2`      |
| linkUrl   | 跳转链接，默认点击整个卡片以 href 的形式跳转，可通过点击事件自定义 | String  | -      |
| imgWidth   | 宽度，默认单位`px` | String  | -      |
| imgHeight   | 高度，默认单位`px` | String  | -      |
| loadingImg   | 设置加载中提示图片，与slotLoding冲突，优先级高于slotLoding | String  | -      |
| errorImg   | 设置错误提示图片，与slotError冲突，优先级高于slotError | String  | -      |


## Events
| 字段 | 说明 | 回调参数 |
|----- | ----- | -----  |
| onClick | 点击事件 |  event: MouseEvent |
