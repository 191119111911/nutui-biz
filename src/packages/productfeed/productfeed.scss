.nut-biz-productfeed {
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  &__main {
    display: flex;
    justify-content: space-between;
    flex-wrap: wrap;
  }
  .nut-biz-card {
    margin-left: 2px;
    &:nth-of-type(2n+1) {
      margin-left: 0;
    }
  }
}
