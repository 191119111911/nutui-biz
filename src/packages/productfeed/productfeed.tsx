import React, {
  FunctionComponent, ReactNode, useEffect, useState
} from "react"
import { Infiniteloading } from "@nutui/nutui-react";
import { IComponent } from "@/utils/typings"

import classNames from "classnames"
import bem from "@/utils/bem"
import { ProductFeedItem } from "./productfeeditem"

export interface ProductFeedProps extends IComponent {
  customProduct: (item: any) => ReactNode
  data: any // Array<any>
  // 是否还有更多数据
  hasMore: boolean
  // 在 useWindow 属性为 false 的时候，自定义设置节点ID
  containerId: string
  // 将滚动侦听器添加到 window 否则侦听组件的父节点
  useWindow: boolean
  // “没有更多数”据展示文案
  loadMoreTxt: string
  // 上拉加载图标名称
  loadIcon: string
  // 上拉加载提示文案
  loadTxt: string
  // 是否开启下拉刷新
  isOpenRefresh: boolean
  // 下拉刷新图标名称
  pullIcon: string
  // 下拉刷新提示文案
  pullTxt: string
  // 继续加载的回调函数
  onLoadMore: (param: () => void) => void
  // 下拉刷新事件回调
  onRefresh: (param: () => void) => void
  // 实时监听滚动高度
  onScrollChange: () => void
  col: number | string
  padding: number | string
  borderRadius: number | string
  imgUrl: string
  imgWidth: string
  imgHeight: string
  imgTag: ReactNode
  isImageLazy: boolean
  loadingImg: string
  errorImg: string
  onClick: (item: any) => void
  onImageClick: (item: any) => void

}

const defaultProps = {
  hasMore: true,
  useWindow: true,
  loadMoreTxt: "哎呀，这里是底部了啦",
  loadTxt: "加载中...",
  isOpenRefresh: false,
  pullTxt: "松手刷新",
  data: [],
  col: 2,
  padding: '10px',
  borderRadius: '8px',
  imgWidth: '150px',
  imgHeight: '150px',
  isImageLazy: true,
  loadingImg: '//img12.360buyimg.com/imagetools/jfs/t1/180776/26/8319/4587/60c094a8E1ef2ec9d/940780b87700b1d3.png',
  errorImg: '//img12.360buyimg.com/imagetools/jfs/t1/180776/26/8319/4587/60c094a8E1ef2ec9d/940780b87700b1d3.png',
  // onClick: () => { },
  // onImageClick: () => { }
} as ProductFeedProps

export const ProductFeed: FunctionComponent<
  Partial<ProductFeedProps> & Omit<React.HTMLAttributes<HTMLDivElement>, 'onClick'>
> = (props) => {
  const {
    className,
    style,
    children,
    data,
    customProduct,
    hasMore,
    containerId,
    useWindow,
    loadMoreTxt,
    loadIcon,
    loadTxt,
    isOpenRefresh,
    pullIcon,
    pullTxt,
    onLoadMore,
    onRefresh,
    onScrollChange,
    col,
    padding,
    borderRadius,
    imgUrl,
    imgWidth,
    imgHeight,
    imgTag,
    isImageLazy,
    loadingImg,
    errorImg,
    onClick,
    onImageClick,
    ...rest
  } = {
    ...defaultProps,
    ...props,
  }
  const b = bem("biz-productfeed")

  const  [listLeft, setListLeft] = useState([] as any)
  const  [listRight, setListRight] = useState([] as any)

  const productItem = (item: any)=>{
    return (
      <ProductFeedItem
        key={item.id}
        data={item}
        col={col}
        imgUrl={item.imgUrl}
        imgWidth={imgWidth}
        imgHeight={imgHeight}
        imgTag={imgTag}
        onClick={onClick}
        onImageClick={onImageClick}
      >
        {customProduct(item)}
      </ProductFeedItem>
    )
  }

  const init = () => {
    const leftLen = listLeft.length
    const rightLen = listRight.length

    if (listLeft.length >= data.length/2 && listRight.length >= data.length/2) {
    } else {
      for (let i = leftLen + rightLen; i < (leftLen + rightLen + 6 > data.length ? data.length : leftLen + rightLen + 6) ; i++) {
        i % 2 == 0 ? listLeft.push(data[i]) : listRight.push(data[i])
      }
      setListLeft([...listLeft])
      setListRight([...listRight])
    }
  }

  useEffect(() => {
    col == 2 && init();
  }, [data])

  return (
    <div className={classNames([b(), className])} style={style} {...rest}>
      <Infiniteloading
        containerId={containerId}
        useWindow={useWindow}
        hasMore={hasMore}
        loadMoreTxt={loadMoreTxt}
        loadIcon={loadIcon}
        loadTxt={loadTxt}
        isOpenRefresh={isOpenRefresh}
        pullIcon={pullIcon}
        pullTxt={pullTxt}
        onLoadMore={onLoadMore}
        onRefresh={onRefresh}
        onScrollChange={onScrollChange}
      >
        <div className={b("main")}>
        { col == 1 ? 
            data.map((item: any)=> {
              return (
                productItem(item)
              )
            }) 
          :
            <>
              <div className={b("left")}>
                { listLeft.map((item: any)=> {
                  return (
                    productItem(item)
                  )
                }) }
              </div>
              <div className={b("right")}>
                { listRight.map((item: any)=> {
                  return (
                    productItem(item)
                  )
                }) }
              </div>
            </>
          }
        </div>
      </Infiniteloading>
    </div>
  )
}

ProductFeed.defaultProps = defaultProps
ProductFeed.displayName = "NutProductFeed"
