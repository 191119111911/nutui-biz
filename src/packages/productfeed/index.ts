import { ProductFeed } from './productfeed'

export default ProductFeed
