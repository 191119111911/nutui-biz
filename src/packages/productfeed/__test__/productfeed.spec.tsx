import * as React from 'react'

import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { trigger, triggerDrag } from '@/utils/test/event'

import { ProductFeed } from '../productfeed'

const intersectionObserverMock = () => ({
  observe: () => null
})
window.IntersectionObserver = jest.fn().mockImplementation(intersectionObserverMock);

test('col test', () => {
  const data = [{
    id: 1,
    imgUrl: "//img13.360buyimg.com/imagetools/jfs/t1/190855/7/12881/42147/60eb0cabE0c3b7234/d523d551413dc853.png",
    name: "我是标题",
    desc: "更多买点",
    price: "388",
  }]
  const { container } = render(
    <ProductFeed
      data={data}
      col={2}
      imgUrl="imgUrl"
      openInfiniteloading={false}
    />
  )

  expect(container.querySelector('.nb-productfeed__left')).toHaveStyle(
    'width: 50%);'
  )
})

test('image test', () => {
  const data = [{
    id: 1,
    imgUrl: "//img13.360buyimg.com/imagetools/jfs/t1/190855/7/12881/42147/60eb0cabE0c3b7234/d523d551413dc853.png",
    name: "我是标题",
    desc: "更多买点",
    price: "388",
  }]
  const { container } = render(
    <ProductFeed
      id="refreshScroll1"
      data={data}
      col={2}
      imgUrl="imgUrl"
      imgWidth="160"
      imgHeight="160"
      imgTag={<div className="img-label"><img src="https://img12.360buyimg.com/imagetools/jfs/t1/186347/7/7338/1009/60c0806bE0b6c7207/97fd04b48d689ffe.png" /></div>}
      openInfiniteloading={false}
    />
  )

  expect(container.querySelector('.nut-img')).toHaveAttribute(
    'data-src',
    '//img13.360buyimg.com/imagetools/jfs/t1/190855/7/12881/42147/60eb0cabE0c3b7234/d523d551413dc853.png'
  )

  expect(container.querySelector('.nut-image')).toHaveAttribute(
    'style',
    'height: 160px; width: 160px;'
  )

  const imgTag = container.querySelectorAll('.img-label')
  expect(imgTag.length).toBe(1)
})

test('onRefresh test', () => {
  const data = [{
    id: 1,
    imgUrl: "//img13.360buyimg.com/imagetools/jfs/t1/190855/7/12881/42147/60eb0cabE0c3b7234/d523d551413dc853.png",
    name: "我是标题",
    desc: "更多买点",
    price: "388",
  }, {
    id: 2,
    imgUrl: "//img13.360buyimg.com/imagetools/jfs/t1/190855/7/12881/42147/60eb0cabE0c3b7234/d523d551413dc853.png",
    name: "我是标题",
    desc: "更多买点",
    price: "388",
  }]
  const refresh = jest.fn()
  const { container } = render(
    <ProductFeed
    id="refreshScroll2"
      data={data}
      col={2}
      imgUrl="imgUrl"
      imgWidth="160"
      imgHeight="160"
      imgTag={<div className="img-label"><img src="https://img12.360buyimg.com/imagetools/jfs/t1/186347/7/7338/1009/60c0806bE0b6c7207/97fd04b48d689ffe.png" /></div>}
      infiniteloadingProps={{
        isOpenRefresh: true,
        onRefresh: refresh
      }}
    />
  )

  const track = container.querySelector('.nut-infiniteloading')
  // pulling
  trigger(track, 'touchstart', 0, 0)
  trigger(track, 'touchmove', 0, 20)
  expect(container).toMatchSnapshot()

  // loading
  trigger(track, 'touchend', 0, 100)
  expect(container).toMatchSnapshot()

  // still loading
  triggerDrag(track, 0, 100)
  expect(refresh).toBeCalled()

})

test('infiniteloading test', () => {
  const App = () => {
    const [list, setList] = React.useState<any[]>([])
    const [hasMore, setHasMore] = React.useState(true)

    React.useEffect(() => {
      init()
    }, [])

    const init = () => {
      for (let i = 0; i < 6; i++) {
        list.push({
          id: i + 1,
          imgUrl: "//img13.360buyimg.com/imagetools/jfs/t1/190855/7/12881/42147/60eb0cabE0c3b7234/d523d551413dc853.png",
          name: "我是标题",
          desc: "更多买点",
          price: "388",
        })
      }
      setList([...list])
    }

    const loadMore = (done: () => void) => {
      setTimeout(() => {
        const curLen = list.length
        for (let i = curLen; i < curLen + 10; i++) {
          list.push(`${i}`)
        }
        if (list.length >= 30) {
          setHasMore(false)
        } else {
          setList([...list])
        }
        done()
      }, 500)
    }
    return (
      <ProductFeed
        className="demo3"
        data={list}
        infiniteloadingProps={{
          hasMore: hasMore,
          onLoadMore: loadMore,
        }}
        col={2}
        imgUrl="imgUrl"
      />
    )
  }
  const { container } = render(<App />)
  const track = container.querySelector('.nut-infiniteloading')
  trigger(track, 'touchstart', 0, 0)
  trigger(track, 'touchmove', 0, -100)
  trigger(track, 'touchend', 0, -800)
  triggerDrag(track, 0, -800)
  expect(container).toMatchSnapshot()
})
