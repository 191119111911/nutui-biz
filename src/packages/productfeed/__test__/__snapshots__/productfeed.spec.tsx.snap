// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`infiniteloading test 1`] = `
<div>
  <div
    class="nb-productfeed demo3"
  >
    <div
      class="nut-infiniteloading"
    >
      <div
        class="nut-infinite-top"
        style="height: 0px; transition: height 0.2s cubic-bezier(0.25,0.1,0.25,1);"
      >
        <div
          class="top-box"
        >
          <img
            class="nut-icon__img top-img "
            src="https://img10.360buyimg.com/imagetools/jfs/t1/169863/6/4565/6306/60125948E7e92774e/40b3a0cf42852bcb.png"
          />
          <span
            class="top-text"
          >
            松开刷新
          </span>
        </div>
      </div>
      <div
        class="nut-infinite-container"
      >
        <div
          class="nb-productfeed__main"
        >
          <div
            class="nb-productfeed__left"
          >
            <div
              class="nb-productfeedItem nb-productfeedItem__multiple"
              style="border-radius: 8px; padding: 10px;"
            >
              <div
                class="nb-productfeedItem__image"
              >
                <div
                  class="nut-image  "
                  style="height: 150px; width: 150px;"
                >
                  <img
                    alt=""
                    class="nut-img lazyload"
                    data-src="//img13.360buyimg.com/imagetools/jfs/t1/190855/7/12881/42147/60eb0cabE0c3b7234/d523d551413dc853.png"
                    loading="lazy"
                    style="object-fit: fill; object-position: center;"
                  />
                  <div
                    class="nut-img-loading"
                    style="background-image: url(//img12.360buyimg.com/imagetools/jfs/t1/180776/26/8319/4587/60c094a8E1ef2ec9d/940780b87700b1d3.png);"
                  />
                </div>
              </div>
              <div
                class="nb-productfeedItem__content"
              />
            </div>
            <div
              class="nb-productfeedItem nb-productfeedItem__multiple"
              style="border-radius: 8px; padding: 10px;"
            >
              <div
                class="nb-productfeedItem__image"
              >
                <div
                  class="nut-image  "
                  style="height: 150px; width: 150px;"
                >
                  <img
                    alt=""
                    class="nut-img lazyload"
                    data-src="//img13.360buyimg.com/imagetools/jfs/t1/190855/7/12881/42147/60eb0cabE0c3b7234/d523d551413dc853.png"
                    loading="lazy"
                    style="object-fit: fill; object-position: center;"
                  />
                  <div
                    class="nut-img-loading"
                    style="background-image: url(//img12.360buyimg.com/imagetools/jfs/t1/180776/26/8319/4587/60c094a8E1ef2ec9d/940780b87700b1d3.png);"
                  />
                </div>
              </div>
              <div
                class="nb-productfeedItem__content"
              />
            </div>
            <div
              class="nb-productfeedItem nb-productfeedItem__multiple"
              style="border-radius: 8px; padding: 10px;"
            >
              <div
                class="nb-productfeedItem__image"
              >
                <div
                  class="nut-image  "
                  style="height: 150px; width: 150px;"
                >
                  <img
                    alt=""
                    class="nut-img lazyload"
                    data-src="//img13.360buyimg.com/imagetools/jfs/t1/190855/7/12881/42147/60eb0cabE0c3b7234/d523d551413dc853.png"
                    loading="lazy"
                    style="object-fit: fill; object-position: center;"
                  />
                  <div
                    class="nut-img-loading"
                    style="background-image: url(//img12.360buyimg.com/imagetools/jfs/t1/180776/26/8319/4587/60c094a8E1ef2ec9d/940780b87700b1d3.png);"
                  />
                </div>
              </div>
              <div
                class="nb-productfeedItem__content"
              />
            </div>
          </div>
          <div
            class="nb-productfeed__right"
          >
            <div
              class="nb-productfeedItem nb-productfeedItem__multiple"
              style="border-radius: 8px; padding: 10px;"
            >
              <div
                class="nb-productfeedItem__image"
              >
                <div
                  class="nut-image  "
                  style="height: 150px; width: 150px;"
                >
                  <img
                    alt=""
                    class="nut-img lazyload"
                    data-src="//img13.360buyimg.com/imagetools/jfs/t1/190855/7/12881/42147/60eb0cabE0c3b7234/d523d551413dc853.png"
                    loading="lazy"
                    style="object-fit: fill; object-position: center;"
                  />
                  <div
                    class="nut-img-loading"
                    style="background-image: url(//img12.360buyimg.com/imagetools/jfs/t1/180776/26/8319/4587/60c094a8E1ef2ec9d/940780b87700b1d3.png);"
                  />
                </div>
              </div>
              <div
                class="nb-productfeedItem__content"
              />
            </div>
            <div
              class="nb-productfeedItem nb-productfeedItem__multiple"
              style="border-radius: 8px; padding: 10px;"
            >
              <div
                class="nb-productfeedItem__image"
              >
                <div
                  class="nut-image  "
                  style="height: 150px; width: 150px;"
                >
                  <img
                    alt=""
                    class="nut-img lazyload"
                    data-src="//img13.360buyimg.com/imagetools/jfs/t1/190855/7/12881/42147/60eb0cabE0c3b7234/d523d551413dc853.png"
                    loading="lazy"
                    style="object-fit: fill; object-position: center;"
                  />
                  <div
                    class="nut-img-loading"
                    style="background-image: url(//img12.360buyimg.com/imagetools/jfs/t1/180776/26/8319/4587/60c094a8E1ef2ec9d/940780b87700b1d3.png);"
                  />
                </div>
              </div>
              <div
                class="nb-productfeedItem__content"
              />
            </div>
            <div
              class="nb-productfeedItem nb-productfeedItem__multiple"
              style="border-radius: 8px; padding: 10px;"
            >
              <div
                class="nb-productfeedItem__image"
              >
                <div
                  class="nut-image  "
                  style="height: 150px; width: 150px;"
                >
                  <img
                    alt=""
                    class="nut-img lazyload"
                    data-src="//img13.360buyimg.com/imagetools/jfs/t1/190855/7/12881/42147/60eb0cabE0c3b7234/d523d551413dc853.png"
                    loading="lazy"
                    style="object-fit: fill; object-position: center;"
                  />
                  <div
                    class="nut-img-loading"
                    style="background-image: url(//img12.360buyimg.com/imagetools/jfs/t1/180776/26/8319/4587/60c094a8E1ef2ec9d/940780b87700b1d3.png);"
                  />
                </div>
              </div>
              <div
                class="nb-productfeedItem__content"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="nut-infinite-bottom"
      />
    </div>
  </div>
</div>
`;

exports[`onRefresh test 1`] = `
<div>
  <div
    class="nb-productfeed"
    id="refreshScroll2"
  >
    <div
      class="nut-infiniteloading"
    >
      <div
        class="nut-infinite-top"
        style="height: 10px; transition: height 0.2s cubic-bezier(0.25,0.1,0.25,1);"
      >
        <div
          class="top-box"
        >
          <img
            class="nut-icon__img top-img "
            src="https://img10.360buyimg.com/imagetools/jfs/t1/169863/6/4565/6306/60125948E7e92774e/40b3a0cf42852bcb.png"
          />
          <span
            class="top-text"
          >
            松开刷新
          </span>
        </div>
      </div>
      <div
        class="nut-infinite-container"
      >
        <div
          class="nb-productfeed__main"
        >
          <div
            class="nb-productfeed__left"
          >
            <div
              class="nb-productfeedItem nb-productfeedItem__multiple"
              style="border-radius: 8px; padding: 10px;"
            >
              <div
                class="nb-productfeedItem__image"
              >
                <div
                  class="nut-image  "
                  style="height: 160px; width: 160px;"
                >
                  <img
                    alt=""
                    class="nut-img lazyload"
                    data-src="//img13.360buyimg.com/imagetools/jfs/t1/190855/7/12881/42147/60eb0cabE0c3b7234/d523d551413dc853.png"
                    loading="lazy"
                    style="object-fit: fill; object-position: center;"
                  />
                  <div
                    class="nut-img-loading"
                    style="background-image: url(//img12.360buyimg.com/imagetools/jfs/t1/180776/26/8319/4587/60c094a8E1ef2ec9d/940780b87700b1d3.png);"
                  />
                </div>
                <div
                  class="nb-productfeedItem__image-tag"
                >
                  <div
                    class="img-label"
                  >
                    <img
                      src="https://img12.360buyimg.com/imagetools/jfs/t1/186347/7/7338/1009/60c0806bE0b6c7207/97fd04b48d689ffe.png"
                    />
                  </div>
                </div>
              </div>
              <div
                class="nb-productfeedItem__content"
              />
            </div>
          </div>
          <div
            class="nb-productfeed__right"
          >
            <div
              class="nb-productfeedItem nb-productfeedItem__multiple"
              style="border-radius: 8px; padding: 10px;"
            >
              <div
                class="nb-productfeedItem__image"
              >
                <div
                  class="nut-image  "
                  style="height: 160px; width: 160px;"
                >
                  <img
                    alt=""
                    class="nut-img lazyload"
                    data-src="//img13.360buyimg.com/imagetools/jfs/t1/190855/7/12881/42147/60eb0cabE0c3b7234/d523d551413dc853.png"
                    loading="lazy"
                    style="object-fit: fill; object-position: center;"
                  />
                  <div
                    class="nut-img-loading"
                    style="background-image: url(//img12.360buyimg.com/imagetools/jfs/t1/180776/26/8319/4587/60c094a8E1ef2ec9d/940780b87700b1d3.png);"
                  />
                </div>
                <div
                  class="nb-productfeedItem__image-tag"
                >
                  <div
                    class="img-label"
                  >
                    <img
                      src="https://img12.360buyimg.com/imagetools/jfs/t1/186347/7/7338/1009/60c0806bE0b6c7207/97fd04b48d689ffe.png"
                    />
                  </div>
                </div>
              </div>
              <div
                class="nb-productfeedItem__content"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="nut-infinite-bottom"
      />
    </div>
  </div>
</div>
`;

exports[`onRefresh test 2`] = `
<div>
  <div
    class="nb-productfeed"
    id="refreshScroll2"
  >
    <div
      class="nut-infiniteloading"
    >
      <div
        class="nut-infinite-top"
        style="height: 10px; transition: height 0.2s cubic-bezier(0.25,0.1,0.25,1);"
      >
        <div
          class="top-box"
        >
          <img
            class="nut-icon__img top-img "
            src="https://img10.360buyimg.com/imagetools/jfs/t1/169863/6/4565/6306/60125948E7e92774e/40b3a0cf42852bcb.png"
          />
          <span
            class="top-text"
          >
            松开刷新
          </span>
        </div>
      </div>
      <div
        class="nut-infinite-container"
      >
        <div
          class="nb-productfeed__main"
        >
          <div
            class="nb-productfeed__left"
          >
            <div
              class="nb-productfeedItem nb-productfeedItem__multiple"
              style="border-radius: 8px; padding: 10px;"
            >
              <div
                class="nb-productfeedItem__image"
              >
                <div
                  class="nut-image  "
                  style="height: 160px; width: 160px;"
                >
                  <img
                    alt=""
                    class="nut-img lazyload"
                    data-src="//img13.360buyimg.com/imagetools/jfs/t1/190855/7/12881/42147/60eb0cabE0c3b7234/d523d551413dc853.png"
                    loading="lazy"
                    style="object-fit: fill; object-position: center;"
                  />
                  <div
                    class="nut-img-loading"
                    style="background-image: url(//img12.360buyimg.com/imagetools/jfs/t1/180776/26/8319/4587/60c094a8E1ef2ec9d/940780b87700b1d3.png);"
                  />
                </div>
                <div
                  class="nb-productfeedItem__image-tag"
                >
                  <div
                    class="img-label"
                  >
                    <img
                      src="https://img12.360buyimg.com/imagetools/jfs/t1/186347/7/7338/1009/60c0806bE0b6c7207/97fd04b48d689ffe.png"
                    />
                  </div>
                </div>
              </div>
              <div
                class="nb-productfeedItem__content"
              />
            </div>
          </div>
          <div
            class="nb-productfeed__right"
          >
            <div
              class="nb-productfeedItem nb-productfeedItem__multiple"
              style="border-radius: 8px; padding: 10px;"
            >
              <div
                class="nb-productfeedItem__image"
              >
                <div
                  class="nut-image  "
                  style="height: 160px; width: 160px;"
                >
                  <img
                    alt=""
                    class="nut-img lazyload"
                    data-src="//img13.360buyimg.com/imagetools/jfs/t1/190855/7/12881/42147/60eb0cabE0c3b7234/d523d551413dc853.png"
                    loading="lazy"
                    style="object-fit: fill; object-position: center;"
                  />
                  <div
                    class="nut-img-loading"
                    style="background-image: url(//img12.360buyimg.com/imagetools/jfs/t1/180776/26/8319/4587/60c094a8E1ef2ec9d/940780b87700b1d3.png);"
                  />
                </div>
                <div
                  class="nb-productfeedItem__image-tag"
                >
                  <div
                    class="img-label"
                  >
                    <img
                      src="https://img12.360buyimg.com/imagetools/jfs/t1/186347/7/7338/1009/60c0806bE0b6c7207/97fd04b48d689ffe.png"
                    />
                  </div>
                </div>
              </div>
              <div
                class="nb-productfeedItem__content"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="nut-infinite-bottom"
      />
    </div>
  </div>
</div>
`;
